### ISSUE TITLE
Post Edit Caching Desync Issue

### ISSUE THREAD

**Issue by cmorwin** [mattermost/mattermost#24165]
#### Summary
Desktop client populates Edit window with an older, cache'd version of a post.

#### Steps to reproduce
Desktop: Version 5.4.0 - commit 420c63d
Mobile: 2.6.0 (Build 479)
Server: 7.8.8 - mysql - Schema 107
1) Create a post on Desktop with "This is my post"
2) Observe "This is my post" on Mobile and Desktop
3) Edit post on Mobile to append ", thanks!"
4) Observe "This is my post, thanks!" on Mobile and Desktop
5) Edit post on Desktop to append " You're welcome!"
6) Observe "This is my post, thanks! You're welcome!" on both Mobile and Desktop
7) Edit post on Mobile to append " Well, this is awkward..."
8) Observe "This is my post, thanks! You're welcome! Well, this is awkward..." on both Mobile and Desktop
9) Edit post on Desktop - Observe the prepopulated text only contains "This is my post, thanks! You're welcome!"
10) Any additional edits to the post from Mobile are viewable on the Desktop, but the Desktop Edit text will not change from "This is my post, thanks! You're welcome!"

This issue was only recently observed - unclear on timing with any potential client/server updates (I don't use the Desktop client consistently). No difference in behavior observed when toggling Settings > Advanced > Allow message drafts to sync with the server. Attached screenshot is in a conversation with myself, but the effect has been observed in public channels as well.

#### Expected behavior
Step 9 should populate with the most current version of the text: "This is my post, thanks! You're welcome! Well, this is awkward..."

#### Observed behavior (that appears unintentional) 
See Reproduction Steps
![Caching_Issue](https://github.com/mattermost/mattermost/assets/119554742/e6a20e01-7e3a-4fe0-87da-a2a9eecba6da)

#### Possible fixes
Not a "fix", but the current workaround I have is to open the post option menu and Copy Text (to preserve formatting), then paste into the edit window before making my changes from Desktop.

**Comment by amyblais** [comment]
Hi @cmorwin, I wasn't able to reproduce this now, but I'm testing on the latest version. Would you have ability to test if you're experiencing the same issue in one of these environments:

 - https://releases.mattermost.com/8.1.0-rc1/mattermost-team-8.1.0-rc1-linux-amd64.tar.gz (RC-1 for upcoming v8.1 ESR release)
 - https://community.mattermost.com/ (on the latest version)

**Comment by amyblais** [comment]
Also, I was testing on MacOS. What OS and version are you on?

**Comment by cmorwin** [comment]
I regularly use the web environment without issue - this only occurs (that I've noticed) on the Windows client. Currently running on Win 10 Pro.

**Comment by amyblais** [comment]
@lindy65 Are you able to reproduce this on Windows on the desktop app?

**Comment by cmorwin** [comment]
Unable to upgrade at this time. I tried reinstalling MM for windows, and it seemed to uninstall and reinstall, although it kept all my server details, so obviously uninstall didn't fully uninstall... Either way, issue still persists. Will update when I'm able to get the server upgraded.

**Comment by amyblais** [comment]
I'll close this for now. We can re-open this if this reproduces on the latest ESR.

**Comment by cmorwin** [comment]
Just updated to 5.6.0 commit aebafea.

When I first clicked the "Edit" option, the box pre-populated with the incorrect/outdated text. After closing the prompt and clicking the "Edit" option again, the correct text was populated instead. I have been able to "Edit" several of my posts (discarding the changes) to fix the weird caching issue.

I am unable to reproduce the issue with this latest version, and any posts I "Edit" don't exhibit the issue after I cancel their first "Edit".
